src/types/Slot0.sol

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+/*
+    Slot0 is a packed version of solidity structure:
+
+    // the current price
+    uint160 sqrtPriceX96;
+    // the current tick
+    int24 tick;
+    // protocol fee, expressed in hundredths of a bip
+    // upper 12 bits are for 1->0, and the lower 12 are for 0->1
+    // the maximum is 1000 - meaning the maximum protocol fee is 0.1%
+    // the protocolFee is taken from the input first, then the lpFee is taken from the remaining input
+    uint24 protocolFee;
+    // used for the lp fee, either static at initialize or dynamic via hook
+    uint24 lpFee;
+*/
+
+type Slot0 is bytes32;
+
+using Slot0Library for Slot0 global;
+
+library Slot0Library {
+    uint160 private constant UINT160_MASK = 0x00FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF;
```

Comment:
why is this not just `0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF` ?

Reply:
Compiler treats it as an address literal and is really angry because of incorrectly formatted address

Reply:
could just change it to `type(uint160).max`? That feels more reliable anyway as you dont need to check you have the right number of Fs

Reply:
> could just change it to `type(uint160).max`?

Only direct number constants can be used in assembly. We can replace it with `type(uin160).max` but it will be a bit messier

Reply:
oh yeah damn, annoying